hcl2template/types.source.go

```diff
@@ -110,6 +110,12 @@ func (cfg *PackerConfig) startBuilder(source SourceUseBlock, ectx *hcl.EvalConte
 	}
 
 	body := source.Body
+	// Add known values to source accessor in eval context.
+	ectx.Variables[sourcesAccessor] = cty.ObjectVal(map[string]cty.Value{
+		"type": cty.StringVal(source.Type),
+		"name": cty.StringVal(source.Name),
+	})
```

Comment:
The following will give the same result and if in the future we add a new attribute we will have to change only in one place.
```suggestion
	ectx.Variables[sourcesAccessor] = cty.ObjectVal(source.ctyValues())
```
The `ctyValues` is also used [here](https://github.com/hashicorp/packer/blob/master/hcl2template/types.packer_config.go#L642)